app/script_variant.py

```diff
@@ -57,12 +69,21 @@ def romanize_japanese_name_by_name_dict(word):
     if not word:
         return ['']
 
-    if word in JAPANESE_NAME_DICTIONARY:
-        yomigana_list = JAPANESE_NAME_DICTIONARY[word]
-        return [jautils.hiragana_to_romaji(yomigana)
-                for yomigana in yomigana_list]
-
-    return [word]
+    words = []
+    for index in xrange(1, len(word)):
+        first_part = word[:index]
+        last_part = word[index:]
+        romanized_first_parts = find_word_in_dict(first_part)
+        romanized_last_parts = find_word_in_dict(last_part)
+        for romanized_first_part in romanized_first_parts:
+            for romanized_last_part in romanized_last_parts:
+                if (romanized_first_part != first_part) and (
```

Comment:
Nit: It would be better to put a single bracket for the whole expression in these two lines. Two brackets are redundant.